zebra-test/tests/transcript.rs

```diff
@@ -2,11 +2,14 @@ use tower::{Service, ServiceExt};
 
 use zebra_test::transcript::Transcript;
 
-const TRANSCRIPT_DATA: [(&str, &str); 4] = [
-    ("req1", "rsp1"),
-    ("req2", "rsp2"),
-    ("req3", "rsp3"),
-    ("req4", "rsp4"),
+type Error = Box<dyn std::error::Error + Send + Sync + 'static>;
+type ErrorChecker = fn(Error) -> Result<(), Error>;
+
+const TRANSCRIPT_DATA: [(&str, Result<&str, ErrorChecker>); 4] = [
```

Comment:
My guess is that we're going to want to just define tests in the test fn so we can use `let` and type inference.

Reply:
type inference was some nightmare shit, it looks like using consts / statics, manually specified types, and fn pointers gives the most ergonomic interface.